src/util/simplify_expr.cpp

```diff
@@ -235,6 +235,50 @@ bool simplify_exprt::simplify_function_application(exprt &expr)
     expr = from_integer(is_prefix ? 1 : 0, expr.type());
     return false;
   }
+  else if(func_id == ID_cprover_string_endswith_func)
+  {
+    const refined_string_exprt &s1 =
```

Comment:
See the remark on the recently merged PR: it would be preferable to factor each of the simplifications into their own local/`static` procedure.

Reply:
Done